src/wasm/wasm-binary.cpp

```diff
@@ -1364,10 +1420,9 @@ void WasmBinaryBuilder::readImports() {
           throwError("invalid event index " + std::to_string(index) + " / " +
                      std::to_string(wasm.functionTypes.size()));
         }
-        Name type = wasm.functionTypes[index]->name;
-        std::vector<Type> params = wasm.functionTypes[index]->params;
+        Type params = Type(wasm.functionTypes[index]->params);
```

Comment:
Just to clarify, we are still using `Module::functionTypes` for 1. bianary reading 2. wast reading 3. wast writing, but not for binary writing, right? And is this gonna be eventually removed, right? If so, I think it'd be better to make this clear in the PR description.